services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/schema/AnnotationsTranslatorImpl.java

```diff
@@ -367,6 +389,54 @@ void writeAnnotationValue(String key, AnnotationsValue value, JSONObject jsonObj
 			jsonObject.put(key, stringToObject(value.getType(), value.getValue().get(0)));
 		}
 	}
+	
+	/* handle with schema */
+	void writeAnnotationsToJSONObjectWithSchema(Annotations toWrite, JSONObject jsonObject, JsonSchema schema) {
+		Map<String, AnnotationsValue> annotationsMap = toWrite.getAnnotations();
+		// first write with the schema (only writes annotations that appear in the schema as arrays)
+		writeAnnotationValueWithSchema(annotationsMap, jsonObject, schema);
+		// write everything else
+		writeAnnotationsToJSONObject(toWrite, jsonObject);
+	}
+	
+	/* only writes annotations that appear in the schema and are defined as arrays */
+	void writeAnnotationValueWithSchema(Map<String, AnnotationsValue> annotationsMap, 
+			JSONObject jsonObject, JsonSchema schema) {
+		// write the top-level schema first because the SubSchemaIterable does not look at it
+		writeAnnotationsValueWithSchemaHelper(schema, annotationsMap, jsonObject);
```

Comment:
if the subschemaiterator included the root this would be handled in one call.  Can we test if adding the root breaks existing uses of the iterator? If it does then maybe a boolean 'includeRoot'.